internal/controllers/plans.go

```diff
@@ -176,61 +176,60 @@ func (s Server) AddPlanQuotaDefault(ctx echo.Context) error {
 	})
 }
 
+type quotaReq struct {
+	Username     string  `json:"user_name"`
+	ResourceName string  `json:"resource_type_name"`
+	QuotaValue   float64 `json:"quota_value"`
+}
+
 func (s Server) AddQuota(ctx echo.Context) error {
-	username := ctx.Param("user_name")
-	if username == "" {
+
+	var quotaReq quotaReq
+	if err := ctx.Bind(&quotaReq); err != nil {
+		return model.Error(ctx, err.Error(), http.StatusBadRequest)
+	}
+
+	if quotaReq.Username == "" {
 		return model.Error(ctx, "invalid username", http.StatusBadRequest)
 	}
-	resourceName := ctx.Param("resource_name")
-	if resourceName == "" {
-		return model.Error(ctx, "invalid resource Name", http.StatusBadRequest)
+	if quotaReq.ResourceName == "" {
+		return model.Error(ctx, "invalid resource name", http.StatusBadRequest)
 	}
-	quotaValue := ctx.Param("quota_value")
-	if quotaValue == "" {
+	if quotaReq.QuotaValue < 0 {
 		return model.Error(ctx, "invalid Quota value", http.StatusBadRequest)
 	}
-	quotaValueFloat, err := ParseFloat(quotaValue)
-	if err != nil {
-		return model.Error(ctx, "invalid Quota Value", http.StatusInternalServerError)
-	}
-	var resource = model.ResourceType{Name: resourceName}
-	err = s.GORMDB.Debug().Find(&resource, "name=?", resourceName).Error
+	resource, err := db.GetResourceTypeByName(s.GORMDB, quotaReq.ResourceName)
 	if err != nil {
-		return model.Error(ctx, "resource Type not found: "+resourceName, http.StatusInternalServerError)
+		return model.Error(ctx, err.Error(), http.StatusInternalServerError)
 	}
```

Comment:
`db.GetResourceTypeByName` will still return `nil` for the resource type if a resource type with the given name doesn't exist. It'll be necessary to check for that condition as well.